stackslib/src/net/connection.rs

```diff
@@ -971,7 +971,7 @@ impl<P: ProtocolFamily> ConnectionInbox<P> {
             // NOTE: it's important that buf not be too big, since up to buf.len()-1 bytes may need
             // to be copied if a message boundary isn't aligned with buf (which is usually the
             // case).
-            let mut buf = [0u8; 4096];
+            let mut buf = [0u8; 65536];
```

Comment:
I'd prefer not to fragment the heap if possible.  However, we do need this to be allocated per-thread at least, since different threads can potentially call this.  The stack size for the p2p thread is 32 MB.